WMF Framework/CharacterSet+LinkParsing.swift

```diff
@@ -7,6 +7,10 @@ extension CharacterSet {
         return NSCharacterSet.wmf_relativePathAndFragmentAllowed()
     }
     
+    public static var ascii: CharacterSet {
+        return CharacterSet(charactersIn: Unicode.Scalar(0)..<Unicode.Scalar(128))
```

Comment:
Update: Ignore my `rfc3986Allowed` suggestion, that character set breaks a lot of the tests. It also breaks if I add in the rfc reserved characters. Let me know if you find another solution to the caret. Otherwise I'm open to leaving this as-is, since iOS 16 is not a problem.

Reply:
One character set that does seem to pass existing tests plus a caret test is

`CharacterSet(charactersIn: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~:/?#[]@!$&'()*+,;=%`.

That is RFC reserved words and unreserved words, plus the percent sign. I'm still a bit unsure about it, I would need to do more testing with it.

Reply:
Nice catch on having to encode carets! Maybe `URLComponents(string:)` can't handle 1. non-ASCII characters and 2. RFC unsafe characters excluding `%`. If that's the case I think your last character set would work since it seems to include every character that can ever have special meaning in a URL, but I'll do some testing to look for exceptions.

Reply:
So "RFC unsafe" doesn't seem to be an official set of characters, but I was referring to the ones in [this list](https://nicolaslule.com/safe-unsafe-reserved-unreserved-characters-in-urls/): "<>%{}|\^` plus space. It's also the set of ASCII symbol characters (plus space), minus RFC reserved and unreserved characters. I'll refer to this set as the "unsafe characters".

From testing URLComponents, adding any of these unsafe characters (not encoded; excluding `%`; in iOS 15.5) to the path of the href passed into it causes URLComponents to be nil. I also checked RFC 3986 and none of the unsafe characters besides percent seems to have any special use within a URL. So it seems that we don't need to leave the unsafe characters unencoded, except percent.

If we assume that we don't need to leave the unsafe (except percent) and non-ASCII characters unencoded, that just leaves the RFC reserved and unreserved characters plus percent, so I don't think there are any more characters that need to be added to your last `CharacterSet`.

Reply:
Thanks for checking! I'm good with having that last `CharacterSet` then. I haven't noticed any issues with it in my testing. I'll merge after that.